### ISSUE TITLE
Ignoring non-Free Software games by game and data license

### ISSUE THREAD

**Issue by shackra** [desura/Desurium#309]
I'm not sure how Desurium manages the game licenses, however, there is people (included myself) that are interested on financially support **freedom-respecting games** on Desura without bothering for those games how use **DRM** or **EULA/Copyright licenses** in both source code and game assets.

Please add an option, in **configure project source code phase**, to complete ignore any game that isn't freedom-respecting, **so Desurium can be provided on any "Free Distro Guideline" compliant GNU/Linux distribution!**

**Comment by karolherbst** [comment]
Isn't it something like #93 ? I don't like the idea, to have a build time switch for that, because the ability of choice us much more important as the enforcement of freedom. Even the history has shown, that any enforcement of freedom was bad.

But it would make sense, to have a runtime switch for that.

**Comment by karolherbst** [comment]
For me, this is going too far. With the same argument you could suggest a blocking mechanism in FOSS web browser, so every propritary website would be blocked.

I want to close this issue, because #93 is enough for me. If anybody is against this, he should leave a comment here.

We could talk about a runtime switch, but not compile time.

**Comment by elzorrorebelde** [comment]
The switch at compiling time is necessary for **FSDG compliant GNU/Linux distributions** if any packager wants to provide Desurium for such GNU/Linux distributions, as Blender have a switch to disable CUDA Toolkit. Is not a issue about ability of choice because we already did.

Please don't dismiss this feature request. #93 is not enough because Desurium will be still providing non-Free games.

**Comment by karolherbst** [comment]
@shackra no, that doesn't make sense. If you want to disallow desurium to show propritary content, you have to disallow this for any webbrowser, too.

CUDA is a "compile time" tookit. Of course you are disabling this in a FSDG compliant GNU/Linux distribution.

All drm related stuff is only stored on the desura server. Desurium isn't providing any runtime binaries for this. So it is completly up to the game devs to implement drm mechanisms

**Comment by elzorrorebelde** [comment]
@karolherbst We already have LibreJS that block non-free non-trivial Javascript code and Gnash to running Flash content on websites for Free Software browsers.

Please, understand **our position and values**. If Desurium provides non-free software it couldn't be added to any **FSDG compliant GNU/Linux distribution**. If game developers add the source code and game data license information we could discern which games respect our freedom and which ones don't, moreover, a switch at compiling time to complete ignore such games that don't respect his user's freedom will make Desurium compatible with any FSDG compliant GNU/Linux distributions.

**Comment by karolherbst** [comment]
First: Desurium does not provide non-free software. The User is deciding to install it.
Second: If you use gnash, you are still able to watch "propritary" or "non-free" content.

For me, desurim is already FSDG compliant, because there is nothing non-free in the code base.

We could discuss here any further and I could bring some points, which will show how impossible it is to block all proprietary content in a webbrowser (like you can't see, if the HTML is generated by a proprietary server back end, etc...)

As I said: we are willing to add some tags like "drm" or "FOSS" and stuff. But for this, we need some changes on the desura server back end. Further we would need some more time to implement stuff like that. Currently we are preparing for our first milestone and a lot of other things have to be done first.

So either you would help us to implement this feature, or you have to wait (and this could take months, not because we aren't able to implement this in a few days, but more, because other things are just more important for us)

**Comment by elzorrorebelde** [comment]
We could wait until those changes at Desura server backend and this feature request were implemented, but please don't close/dismiss this feature request at all.

thank you.

**Comment by csolisr** [comment]
What the user above wants is to release a version that filters all non-free games, _NOT_ to use it as the main version. Just ship an _optional_ filter and point users of free-as-in-freedom GNU+Linux-Libre distros to it.

**Comment by karolherbst** [comment]
and for me this sounds like censorship. I don't want to steal the possibility to install non-free content from the user.

As I said above: on the most websites, you can't see, if this site is generated by propritary software, so we have the same problem there. Of course you are able to filter all non-free websites, but how do you want to create such a hardcoded filter, without implementing censorship?

Further I said: We could tag every game, further we could implement a "runtime" switch for that. If a runtime switch is not enough, then every webbrowser is not FSDG compliant, because they are allowing to display non-free content.

**Comment by karolherbst** [comment]
And because I am against censorship, you won't get my support to implement hardcoded compile time switches for this. If you want to write a patch, then do it and publish this patch.

But I don't like it, when desurium begins to incapacitate its users. This should not be the goal

**Comment by adrianheine** [comment]
As far as I see it, Desurium is DFSG-compatible even when it allows to install non-free software. After all, dpkg allows to install non-free software as well, and apt-get even actively supports this when configured correctly.

**Comment by LiamDawe** [comment]
Also don't forget Desura is a company selling games.

I agree with Karolherbst and adrianlang, i think personally shackra is being ridiculous.

**Comment by csolisr** [comment]
Not because most distros ship non-free software means that distros that decide to ship free software only don't have the right to keep doing so. The patch requested is meant to be optional and exclusive to FSF-endorsed distros. About web browsers that display non-free content, the FSF recently released an addon, LibreJS, that actively blocks JavaScript code that is considered as non-free, but it's exclusive to FSF-endorsed distros. Users that desire to use free software only, as I pointed earlier, deserve to have that choice respected. The solution would be an alternate branch (let's call it desurium-libre to keep with the naming used by Parabola GNU+Linux-Libre) that filtered all non-free games and kept free-as-in-freedom games only, similarly to what GNU IceCat does with addons, by keeping a list that consists only on free software addons.

**Comment by LiamDawe** [comment]
I agree an option would be nice but seriously this is supposed to be a game client for a company that sells games please don't forget that, there must be better places like your friendly repos for free games.

The devs are already snowed under with far more important tasks right now!

**Comment by csolisr** [comment]
Don't worry - Parabola's devs can host and maintain the liberated repos. The only thing we need is the method to patch the list of games available.

**Comment by Mailaender** [comment]
"Desura is a DRM agnostic" http://deltagamer.com/12143/interview-desuras-dave-traeger-talks-digital-distribution I doubt they will change this philosophy. Shipping DRM software does not make it non-free. Even RMS says that non-free games are not optimal, but the availability has a greater usage to the GNU/Linux ecosystem. Also game art does not need to be freely licensed according to St. Ignutius. http://www.gnu.org/philosophy/nonfree-games.html As DRM-free is an extra benefit maybe some publisher will advertise this on their own.

You should install free/libre open source software games using distributions package manager by the way as Desurium will also install dependencies into your home that you already have on your system or maybe a system library will not be found (like libcurl in Urban Terror).
